akka-actor-tests/src/test/scala/akka/util/ByteStringSpec.scala

```diff
@@ -364,14 +361,49 @@ class ByteStringSpec extends WordSpec with Matchers with Checkers {
       ByteStrings(ByteString1.fromString("ab"), ByteString1.fromString("")).drop(Int.MinValue) should ===(ByteString("ab"))
       ByteStrings(ByteString1.fromString(""), ByteString1.fromString("ab")).dropRight(Int.MinValue) should ===(ByteString("ab"))
     }
-    "slice" in {
-      ByteStrings(ByteString1.fromString("a"), ByteString1.fromString("")).slice(0, 1) should ===(ByteString("a"))
+    "slice " in {
       ByteStrings(ByteString1.fromString(""), ByteString1.fromString("a")).slice(1, 1) should ===(ByteString(""))
+      // Can obtain expected results from 6 basic patterns
```

Comment:
We have ScalaChecks, and if it finds an edge case that I for sure what to cover,
that totally makes sense to put into an explicit test. Esp. because this is exactly how we missed this bug: "oh yeah we have scalacheck!" which gave us false confidence.

Only after running 1000s times (instead of default settings) ScalaCheck would get into these edge cases. So I want them to be specifically covered here, in addition to SC.